File: app/src/main/java/com/codemybrainsout/imageviewer/view/search/SearchModule.java
package com.codemybrainsout.imageviewer.view.search;

import android.content.Context;

import com.codemybrainsout.imageviewer.api.SearchService;
import com.codemybrainsout.imageviewer.dagger.scope.ActivityScope;
import com.codemybrainsout.imageviewer.view.home.MultiViewAdapter;
import com.codemybrainsout.imageviewer.view.photo.PhotoAdapter;

import dagger.Module;
import dagger.Provides;

/**
 * Created by ahulr on 11-06-2017.
 */

@Module
public class SearchModule {

    private Context context;

    public SearchModule(Context context) {
        this.context = context;
    }

    @ActivityScope
    @Provides
    Context context() {
        return context;
    }

    @ActivityScope
    @Provides
    MultiViewAdapter multiViewAdapter(Context context) {
        return new MultiViewAdapter(context);
    }

    @ActivityScope
    @Provides
    SearchPresenter searchPresenter(SearchService searchService) {
        return new SearchPresenter(searchService, (SearchContract.View) context);
    }
}

File: app/src/main/java/com/codemybrainsout/imageviewer/view/search/SearchModule.kt
package com.codemybrainsout.imageviewer.view.search

import dagger.Binds
import dagger.Module
import dagger.hilt.InstallIn
import dagger.hilt.android.components.ActivityComponent

/**
 * Created by ahulr on 11-06-2017.
 */
@Module
@InstallIn(ActivityComponent::class)
abstract class SearchModule {

    @Binds
    abstract fun bindActivity(activity: SearchActivity): SearchContract.View

    @Binds
    abstract fun bindPresenter(impl: SearchPresenter): SearchContract.Presenter

}